src/components/PaymentCycleLauncher.js

```diff
@@ -0,0 +1,151 @@
+import React, { useState, useEffect, useRef } from 'react';
+import { connect } from 'react-redux';
+import { makeStyles } from '@material-ui/core/styles';
+import { bindActionCreators } from 'redux';
+import {
+  Paper, Grid, Divider, IconButton,
+} from '@material-ui/core';
+import {
+  FormattedMessage,
+  useTranslations,
+  useModulesManager,
+  PublishedComponent,
+  coreConfirm,
+  clearConfirm,
+  journalize,
+} from '@openimis/fe-core';
+import SendIcon from '@material-ui/icons/Send';
+import { generatePaymentCycle } from '../actions';
+
+const useStyles = makeStyles((theme) => ({
+  paper: {
+    marginBottom: theme.spacing(1),
+  },
+  paperHeader: theme.paper.header,
+  paperHeaderTitle: theme.paper.title,
+  paperHeaderAction: theme.paper.action,
+  form: {
+    padding: 0,
+  },
+  item: {
+    padding: theme.spacing(1),
+  },
+  paperDivider: theme.paper.divider,
+}));
+function PaymentCycleLauncher({
+  generatePaymentCycle,
+  confirmed,
+  coreConfirm,
+  clearConfirm,
+  mutation,
+  submittingMutation,
+}) {
+  const modulesManager = useModulesManager();
+  const classes = useStyles();
+  const { formatMessage, formatMessageWithValues } = useTranslations('paymentCycle', modulesManager);
+  const prevSubmittingMutationRef = useRef();
+  const [filters, setFilters] = useState({});
+  const min = new Date().getFullYear() - 7;
+  const max = min + 9;
```

Comment:
can we add '9' to the constanst? The same for '7' line above. It might be useful to understand the meaning of particular values. At first glance it is not clear for me.

Reply:
done